### PR TITLE
TAUR-1271 TAUR-851 Update ami bake

### PR DESCRIPTION
**Pull request by unixorn**
* Add openssl-dev to AMIs
* Configure salt to use our salt master without us monkeying with '/etc/resolv.conf`
* Preset minion id so salt's bootstrap.sh doesn't attempt to use the default one that isn't preset to be approved on the master
* Turn on debug mode in `bootstrap.sh` so we can see details when it fails
* Add mysql-python to the AMI to try and fix TAUR-851's salt double-run problem
* We only want to purge authorized_keys on grok AMIs so we can pass marketplace acceptance, not every AMI.

### CODE REVIEW

**Comment by unixorn** [comment]
@shantanoo @jcasner please CR

File: infrastructure/ami-tools/packer-scripts/cleanup-image [hunk 1]
```diff
@@ -100,3 +82,20 @@ df -h /
...
+complain "Zapping logfiles..."
+logger -t image-cleanup "Zapping logfiles..."
+for logf in /var/log/cron \
+            /var/log/dracut.log \
+            /var/log/maillog \
+            /var/log/messages \
+            /var/log/secure \
+            /var/log/yum.log
+do
+  complain "Resetting ${logf}"
```
Comment (jcasner) [inline infrastructure/ami-tools/packer-scripts/cleanup-image]: As long as we're in here, can we rename `complain` to `log` or something similar?

File: infrastructure/ami-tools/packer-scripts/configure-infrastructure-ami [hunk 2]
```diff
@@ -55,38 +65,21 @@ cat /etc/salt/minion_id
...
-grep ^master: /etc/salt/minion
-retcode="$?"
-if [ "${retcode}" != "0" ]; then
-  echo "Failed to set salt master name"
-  echo "Check for changes to default /etc/salt/minion"
-  exit "${retcode}"
-fi
+# echo "Setting search list in /etc/resolv.conf to numenta.com"
+# # sed 's/^search .*/search numenta.com/g' /etc/resolv.conf -i''
+# cat /etc/resolv.conf
```
Comment (jcasner) [inline infrastructure/ami-tools/packer-scripts/configure-infrastructure-ami]: Remove commented code

File: infrastructure/ami-tools/packer-scripts/configure-infrastructure-ami [hunk 3]
```diff
@@ -55,38 +65,21 @@ cat /etc/salt/minion_id
...
+# host salt.numenta.com
 
-echo "Purging salt minion key data..."
-rm -fr /etc/salt/pki/minion/*
+echo "Salt minion_id: $(cat /etc/salt/minion_id)"
 
-echo "Restarting salt minion..."
-service salt-minion restart
+# echo "Purging salt minion key data..."
+# rm -fr /etc/salt/pki/minion/*
```
Comment (jcasner) [inline infrastructure/ami-tools/packer-scripts/configure-infrastructure-ami]: Remove commented code

**Comment by unixorn** [comment]
Ready for another CR, @jcasner

File: infrastructure/ami-tools/packer-scripts/install-salt [hunk 4]
```diff
@@ -67,6 +73,7 @@ prep_instance_for_salt(){
 setup_directories(){
   mkdir -p /usr/local/sbin
   mkdir -p /usr/local/bin
+  mkdir -p /etc/salt
```
Comment (shantanoo) [inline infrastructure/ami-tools/packer-scripts/install-salt]: Use sorted sequence for `mkdir`. Would prefer using `for`.

e.g.

```
for directory in /etc/salt /usr/local/{,s}bin
do
    mkdir -p "${directory}"
done
```

**Comment by shantanoo** [comment]
LGTM.

**Pull request merged**